src/controllers/controller.bar.js

```diff
@@ -95,6 +95,92 @@ defaults._set('horizontalBar', {
 	}
 });
 
+/**
+ * Computes the "optimal" sample size to maintain bars equally sized while preventing overlap.
+ * @private
+ */
+function computeMinSampleSize(scale, pixels) {
+	var min = scale.isHorizontal() ? scale.width : scale.height;
+	var ticks = scale.getTicks();
+	var prev, curr, i, ilen;
+
+	for (i = 0, ilen = pixels.length; i < ilen; ++i) {
+		min = i > 0 ? Math.min(min, pixels[i] - pixels[i - 1]) : min;
```

Comment:
I think we can start this loop at 1 and then the ternary condition goes away

Reply:
Good catch, that was originally part of [this loop](/chartjs/Chart.js/pull/4994/files#diff-5d42034a6fdb9268bcdfd72b4ee5e34fR358) (that's why `i > 0`), but forgot after moving it into a separate method.